static/js/components/EducationForm.js

```diff
@@ -98,14 +102,60 @@ class EducationForm extends ProfileFormFields {
     </Grid>;
   };
 
-  handleSwitchClick(level){
+  entriesForLevel: Function = (level: string): number[] => {
+    const { profile } = this.props;
+    if ( _.isEmpty(profile.education) ) {
+      return [];
+    } else {
+      return Object.entries(profile.education).
+        filter(([, entry]: [string, EducationEntry]) => entry.degree_name === level).
+        map(([index,]) => Number(index));
+    }
+  };
+
+  closeDeleteAllEducationDialog: Function = (): void => {
     const {
-      ui: { educationDegreeInclusions },
+      setEducationDegreeLevel,
+      setShowEducationDeleteAllDialog,
+    } = this.props;
+    setEducationDegreeLevel('');
+    setShowEducationDeleteAllDialog(false);
+  };
+
+  deleteAllEducationEntriesForLevel: Function = (): void => {
+    const {
+      profile,
+      saveProfile,
+      ui,
+      ui: { educationDegreeLevel, educationDegreeInclusions },
       setEducationDegreeInclusions,
     } = this.props;
+    let clone = _.cloneDeep(profile);
+    clone.education = clone.education.filter(entry => (
+      entry.degree_name !== educationDegreeLevel
+    ));
     let newState = Object.assign({}, educationDegreeInclusions);
-    newState[level] = !educationDegreeInclusions[level];
+    newState[educationDegreeLevel] = false;
     setEducationDegreeInclusions(newState);
+    saveProfile(educationValidation, clone, ui);
+  };
+
+  handleSwitchClick(level: string): void {
+    const {
+      ui: { educationDegreeInclusions },
+      setEducationDegreeInclusions,
+      setEducationDegreeLevel,
+      setShowEducationDeleteAllDialog,
+    } = this.props;
+    if ( _.isEmpty(this.entriesForLevel(level)) ) {
```

Comment:
Yes, I was previously using `entriesForLevel` for deleting entries when removing a degree level, but now that we're not using it there it can probably be deleted.